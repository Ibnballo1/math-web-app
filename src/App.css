@import url('https://fonts.googleapis.com/css2?family=Lato&family=Montserrat+Subrayada:wght@400;700&family=Poppins:ital,wght@0,400;0,500;0,600;0,700;1,600&display=swap');

.section {
  display: flex;
  justify-content: flex-end;
  align-items: center;
  height: 97vh;
}

.container {
  display: grid;
  grid-template-columns: repeat(4, auto);
  width: 30%;
  height: 30%;
  font-family: 'Poppins', sans-serif;
  font-size: 12px;
  color: #000;
}

.items {
  background-color: #e0e0e0;
  border: 1px solid #bdbdbd;
  text-align: center;
  padding: 0.75rem;
  margin: 0;
}

.item0 {
  background-color: #858693;
  color: #fff;
  padding-right: 0.5rem;
  font-size: 16px;
  grid-column: 1/5;
  text-align: right;
}

.item17 {
  grid-column: 1/3;
}

.item4,
.item8,
.item12,
.item16,
.item19 {
  background-color: #f5913e;
}

/* .App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
} */
