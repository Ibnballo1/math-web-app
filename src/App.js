/* eslint-disable react/prefer-stateless-function */
import React from 'react';
import Calculator from './components/Calculator';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.handleCalculation = this.handleCalculation.bind('this');
    this.state = {
      total: '5',
      next: '',
      operation: '',
    };
  }

  handleCalculation(newTotal) {
    const { total, next, operation } = this.state;
    console.log(`Total is ${total}, next is ${next} & operation is ${operation}`);
    this.setState({ total: newTotal });
  }

  render() {
    const { total } = this.state;
    return (
      <div className="App">
        <Calculator
          result={total}
          handleCalculation={this.handleCalculation}
        />
      </div>
    );
  }
}

export default App;
