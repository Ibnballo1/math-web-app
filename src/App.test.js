import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render, cleanup } from '@testing-library/react';
import App from './App';

afterEach(cleanup);

it('App component rendered correctly', () => {
  const wrapper = render(
    <MemoryRouter>
      <App />
    </MemoryRouter>,
  );
  expect(wrapper).toMatchSnapshot();
});
