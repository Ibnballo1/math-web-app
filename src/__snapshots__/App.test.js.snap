// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component rendered correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <header
          class="header"
        >
          <h1
            class="headerTitle"
          >
            Math Magicians
          </h1>
          <nav
            class="navbar"
          >
            <a
              class="navLinks"
              href="/"
            >
              Home
            </a>
            <a
              class="navLinks"
              href="/calculator"
            >
              Calculator
            </a>
            <a
              class="navLinks"
              href="/quote"
            >
              Quote
            </a>
          </nav>
        </header>
        <section
          class="homeSection"
        >
          <h1
            class="h1"
          >
            Welcome to our page!
          </h1>
          <p
            class="p"
          >
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Praesentium omnis officia blanditiis laborum necessitatibus deserunt iusto, sapiente ipsa quibusdam expedita! Optio dolores sapiente esse rem doloribus voluptatibus aliquid repellat iure, reprehenderit totam, earum culpa. Nulla necessitatibus incidunt sunt culpa provident quo neque quae officiis.
          </p>
          <p
            class="p"
          >
            Voluptates libero cum totam dolore facilis laborum consequatur explicabo minus tempore exercitationem commodi sunt, culpa officiis illo quaerat soluta suscipit alias quas error nam, ad at esse? Consequatur voluptatem atque, architecto suscipit dolor soluta earum. Repellat nihil sapiente nobis hic ipsam, suscipit quos repellendus dignissimos cupiditate? Culpa autem repellendus voluptatibus repellat nemo eos qui reprehenderit fugit nulla, praesentium ut consequatur dicta delectus molestias debitis commodi possimus temporibus ea libero. Placeat consectetur mollitia iste eligendi, illo sequi quod ipsam minima debitis magni provident voluptates aspernatur quo inventore doloribus omnis accusantium. Itaque sapiente rem enim harum eum recusandae, perferendis deleniti quam ex, tenetur maiores facilis. Mollitia blanditiis tempore illo. Voluptatibus. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Molestias dolor error ut eius, saepe illum, quibusdam sunt accusantium officia sint corporis? Sapiente asperiores explicabo blanditiis in, beatae pariatur iusto architecto?
          </p>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <header
        class="header"
      >
        <h1
          class="headerTitle"
        >
          Math Magicians
        </h1>
        <nav
          class="navbar"
        >
          <a
            class="navLinks"
            href="/"
          >
            Home
          </a>
          <a
            class="navLinks"
            href="/calculator"
          >
            Calculator
          </a>
          <a
            class="navLinks"
            href="/quote"
          >
            Quote
          </a>
        </nav>
      </header>
      <section
        class="homeSection"
      >
        <h1
          class="h1"
        >
          Welcome to our page!
        </h1>
        <p
          class="p"
        >
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Praesentium omnis officia blanditiis laborum necessitatibus deserunt iusto, sapiente ipsa quibusdam expedita! Optio dolores sapiente esse rem doloribus voluptatibus aliquid repellat iure, reprehenderit totam, earum culpa. Nulla necessitatibus incidunt sunt culpa provident quo neque quae officiis.
        </p>
        <p
          class="p"
        >
          Voluptates libero cum totam dolore facilis laborum consequatur explicabo minus tempore exercitationem commodi sunt, culpa officiis illo quaerat soluta suscipit alias quas error nam, ad at esse? Consequatur voluptatem atque, architecto suscipit dolor soluta earum. Repellat nihil sapiente nobis hic ipsam, suscipit quos repellendus dignissimos cupiditate? Culpa autem repellendus voluptatibus repellat nemo eos qui reprehenderit fugit nulla, praesentium ut consequatur dicta delectus molestias debitis commodi possimus temporibus ea libero. Placeat consectetur mollitia iste eligendi, illo sequi quod ipsam minima debitis magni provident voluptates aspernatur quo inventore doloribus omnis accusantium. Itaque sapiente rem enim harum eum recusandae, perferendis deleniti quam ex, tenetur maiores facilis. Mollitia blanditiis tempore illo. Voluptatibus. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Molestias dolor error ut eius, saepe illum, quibusdam sunt accusantium officia sint corporis? Sapiente asperiores explicabo blanditiis in, beatae pariatur iusto architecto?
        </p>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
